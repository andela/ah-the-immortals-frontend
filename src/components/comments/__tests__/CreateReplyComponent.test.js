import React from 'react';
import { Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { createBrowserHistory } from 'history';
import CreateReply from '../CreateReplyComponent';

describe('Tests CommentsReplyComponent', () => {
  const history = createBrowserHistory();
  const props = {
    replyWrite: jest.fn(),
    replySubmit: jest.fn(),
    value: ''
  };
  const wrapper = mount(
    <Router history={history}>
      <CreateReply {...props} />
    </Router>
  );
  it('tests mounting of component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
