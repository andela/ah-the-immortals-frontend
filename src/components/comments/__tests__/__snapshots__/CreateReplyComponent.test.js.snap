// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests CommentsReplyComponent tests mounting of component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <CreateReply
    replySubmit={[MockFunction]}
    replyWrite={[MockFunction]}
    value=""
  >
    <Form
      inline={false}
      onSubmit={[MockFunction]}
    >
      <form
        className=""
        onSubmit={[MockFunction]}
      >
        <Bootstrap(Row)>
          <Row
            bsPrefix="row"
            noGutters={false}
          >
            <div
              className="row"
            >
              <Col>
                <div
                  className="col"
                >
                  <FormControl
                    className="input-width"
                    name="body"
                    onChange={[MockFunction]}
                    placeholder="Leave a reply"
                    required={true}
                    value=""
                  >
                    <input
                      className="input-width form-control"
                      name="body"
                      onChange={[MockFunction]}
                      placeholder="Leave a reply"
                      required={true}
                      value=""
                    />
                  </FormControl>
                </div>
              </Col>
              <Col
                lg="2"
                xs={true}
              >
                <div
                  className="col-lg-2 col"
                >
                  <Button
                    active={false}
                    disabled={false}
                    type="submit"
                    variant="primary"
                  >
                    <button
                      className="btn btn-primary"
                      disabled={false}
                      type="submit"
                    >
                      Send
                    </button>
                  </Button>
                </div>
              </Col>
            </div>
          </Row>
        </Bootstrap(Row)>
      </form>
    </Form>
  </CreateReply>
</Router>
`;
